master/internal/rm/kubernetesrm/spec_test.go

```diff
@@ -221,6 +222,31 @@ func TestAllPrintableCharactersInEnv(t *testing.T) {
 	require.Contains(t, actual, k8sV1.EnvVar{Name: "func", Value: "f(x)=x"})
 }
 
+func TestValidatePodLabelValues(t *testing.T) {
+	tests := []struct {
+		name   string
+		input  string
+		output string
+	}{
+		{"valid all alpha", "simpleCharacters", "simpleCharacters"},
+		{"valid all alphanumeric", "simple4Characters", "simple4Characters"},
+		{"valid contains non-alphanumeric", "simple-Characters.With_Other", "simple-Characters.With_Other"},
+		{"invalid chars", "letters contain *@ other chars -=%", "letters_contain____other_chars"},
+		{"invalid leading chars", "-%4-simpleCharacters0", "4-simpleCharacters0"},
+		{"invalid trailing chars", "simple-Characters4%-.#", "simple-Characters4"},
+		{"invalid too many chars", "simpleCharactersGoesOnForWayTooLong36384042444648505254565860-_AndThenSome", "simpleCharactersGoesOnForWayTooLong36384042444648505254565860"},
+		{"invalid email-style input", "name@domain.com", "name_domain.com"},
+		{"invalid chars only", "-.*$%#$...", ""},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.input, func(t *testing.T) {
+			testOutput := validatePodLabelValue(tt.input)
+			assert.Equal(t, tt.output, testOutput, tt.name+" failed")
```

Comment:
maybe use require.Equal here? I'm not sure what our team's stance is on either package, and tbh I'm not quite sure on the pros/cons of either apart from require being an efficient wrapper on assert

Reply:
I know we usually prefer `require` over `assert`. I can change it. A bit on the pros/cons...

Using `require` is useful when checking dependencies in a test because tests will stop at the first failed require statement. For example, if checking the output of a function:
```
output, err := myFunction(input)
require.NoError(err)
// now we can check the contents of output
```

Using `assert` is useful when checks are fairly independent and it's helpful to have all the information about failed tests because a failed assert will not stop tests.